### DISCUSSION TITLE
.Net Aspire Project structure and Low level understanding about its working

### DISCUSSION THREAD

**Discussion by ks1990cn** [microsoft/aspire#1785] [Q&A] [answered]
Hi,

I need little help to understand the code level and low level architecture of project. Any reference for system design is there from which I can understand this project code and basic fundamentals which are used to develop this?

Should I just start directly debugging from Aspire.Dashboard and Aspire.Hosting?

Asking just understanding so I can start contribution to this repo, it will quite helpful for me 🆘

**Comment by ks1990cn** [comment]
Just got this video from community standup, maybe it can help...
https://www.youtube.com/live/KEcUfMbCgpA?si=A9ip1bO0TBoYHjVZ

Still open for any other insights, if anyone can give.

**Comment by davidfowl** [comment, marked as answer]
You'd have to read the code, the project is being actively developed and changed. I don't expect you'll find any over arching design while that's happening. Happy to answer any questions you have once you've taken a look at the overall project. At a high level there's a dashboard (which is being redesigned), the application model (aspire.hosting), the orchestrator process (dcp), not currently open sourced, and aspire components.

**Reply by ks1990cn** [reply]
Cool, will start understanding. Will post my queries going ahead. Thank you!
